src/ice_shelf/MOM_ice_shelf.F90

```diff
@@ -772,26 +805,43 @@ end subroutine change_thickness_using_melt
 
 !> This subroutine adds the mechanical forcing fields and perhaps shelf areas, based on
 !! the ice state in ice_shelf_CS.
-subroutine add_shelf_forces(G, US, CS, forces, do_shelf_area)
-  type(ocean_grid_type), intent(inout) :: G    !< The ocean's grid structure.
+subroutine add_shelf_forces(US, CS, forces_in, do_shelf_area, external_call)
   type(unit_scale_type), intent(in)    :: US   !< A dimensional unit scaling type
   type(ice_shelf_CS),    pointer       :: CS   !< This module's control structure.
-  type(mech_forcing),    intent(inout) :: forces !< A structure with the driving mechanical forces
+  type(mech_forcing),    pointer       :: forces_in !< A structure with the driving mechanical forces
   logical, optional,     intent(in)    :: do_shelf_area !< If true find the shelf-covered areas.
-
+  logical, optional,     intent(in)    :: external_call !< If true the incoming forcing type
+                                               !! is using the input grid metric and needs
+                                               !! to be rotated.
+  type(ocean_grid_type), pointer :: G => NULL()   !< A pointer to the ocean grid metric.
+  type(mech_forcing),    pointer :: forces => NULL() !< A structure with the driving mechanical forces
   real :: kv_rho_ice ! The viscosity of ice divided by its density [L4 T-1 R-1 Z-2 ~> m5 kg-1 s-1].
   real :: press_ice  ! The pressure of the ice shelf per unit area of ocean (not ice) [R L2 T-2 ~> Pa].
   logical :: find_area ! If true find the shelf areas at u & v points.
+  logical :: rotate = .false.
   type(ice_shelf_state), pointer :: ISS => NULL() ! A structure with elements that describe
                                           ! the ice-shelf state
 
   integer :: i, j, is, ie, js, je, isd, ied, jsd, jed
+
+  if (present(external_call)) rotate=external_call
+
+  if (CS%rotate_index .and. rotate) then
+     allocate(forces)
+     call allocate_mech_forcing(forces_in, CS%Grid, forces)
+     call rotate_mech_forcing(forces_in, CS%turns, forces)
+  else
+     forces=>forces_in
+  endif
+
+  G=>CS%Grid
+
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec
   isd = G%isd ; jsd = G%jsd ; ied = G%ied ; jed = G%jed
 
-  if ((CS%grid%isc /= G%isc) .or. (CS%grid%iec /= G%iec) .or. &
-      (CS%grid%jsc /= G%jsc) .or. (CS%grid%jec /= G%jec)) &
-    call MOM_error(FATAL,"add_shelf_forces: Incompatible ocean and ice shelf grids.")
+!  if ((CS%grid_in%isc /= G%isc) .or. (CS%grid_in%iec /= G%iec) .or. &
```

Comment:
Delete or uncomment? Why would this test not be valid now?

Reply:
resolved. The grid checks are not necessary here any longer. If this is being called from the driver, the fluxes need rotation, otherwise if it is being called from within the ice shelf module, they do not require rotation.